runtime/pkg/duration/timeutil.go

```diff
@@ -0,0 +1,71 @@
+package duration
+
+import (
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+
+	// Load IANA time zone data
+	_ "time/tzdata"
+)
+
+func TruncateTime(start time.Time, tg runtimev1.TimeGrain, tz *time.Location, firstDay, firstMonth int) time.Time {
```

Comment:
Since this doesn't use the `Duration` type at all, should it maybe be moved to a separate package `timeutil`? (Could define the generic non-proto grain type we need in there and use it in the `duration` package)